src/Microsoft.OData.Edm/Schema/EdmModelBase.cs

```diff
@@ -173,25 +198,21 @@ public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(IEdmType b
         /// </returns>
         public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(string qualifiedName, IEdmType bindingType)
         {
-            IEnumerable<IEdmOperation> enumerable = this.FindDeclaredOperations(qualifiedName);
-            IList<IEdmOperation> operations = enumerable as IList<IEdmOperation>;
+            IList<IEdmOperation> operations = this.FindDeclaredBoundOperations(bindingType) as IList<IEdmOperation>;
+            IList<IEdmOperation> matchedOperation = new List<IEdmOperation>();

```

Comment:
>new List<IEdmOperation>(); [](start = 52, length = 26)

rather than create the list here (and allocate the default size which may not be used in the common case where there are no bound operations) allocate the list after checking if operations != null, and add an else{return new IEdmOperation[]{};}